mycity/mycity/intents/snow_parking_intent.py

```diff
@@ -20,49 +18,64 @@
 BOSTON_DATA_PARKING_ADDRESS_INDEX = 9
 
 
-def get_snow_emergency_parking_intent(intent, session):
+def get_snow_emergency_parking_intent(mcd):
+    """
+    Populate MyCityDataModel with snow emergency parking response information.
+
+    :param mcd:
+    :return:
+    """
+    print(
+        '[method: get_snow_emergency_parking_intent]',
+        'MyCityDataModel received:',
+        str(mcd)
+    )
 
-    if alexa_constants.CURRENT_ADDRESS_KEY in session.get('attributes', {}):
+    if intent_constants.CURRENT_ADDRESS_KEY in mcd.session_attributes:
 
-        origin_address = _build_origin_address(session)
+        origin_address = _build_origin_address(mcd)
 
         print("Finding snow emergency parking for {}".format(origin_address))
 
         parking_address, driving_distance, driving_time = \
             _get_snow_emergency_parking_location(origin_address)
 
         if not parking_address:
-            speech_output = "Uh oh. Something went wrong!"
+            mcd.output_speech = "Uh oh. Something went wrong!"
         else:
-            speech_output = \
+            mcd.output_speech = \
                 "The closest snow emergency parking location is at " \
                 "{}. It is {} away and should take you {} to drive " \
                 "there".format(parking_address, driving_distance, driving_time)
 
-        session_attributes = session.get('attributes', {})
-        should_end_session = True
+        mcd.should_end_session = False

```

Comment:
Added an issue to explore different approaches to maintaining the session vs closing it. We should look into best practices on this, but in the meantime I'm leaving this as is because it makes testing less annoying.